### ISSUE TITLE
scrolling the map results in unexpected behavior

### ISSUE THREAD

**Issue by frankystone** [widelands/widelands#5500]
Just updated to last commit and found this behavior:

When scrolling the map sometimes it feels like the right mouse button gets not released: Although the right mouse button is not pressed the map moves with each mouse move. Nothing can be clicked. In this state pressing the right mouse button, to maybe "release" the button again, does not help immediately. After some tries the button get "released" and all is working again, until the next "locked" right mouse button.
In the state of "locked" button, it seems like the mouse pointer get behind the fullscreen of widelands, and something is going on. E.g. a desktop widget get suddenly opened although i never used that widget. Just saw this after successfully closing widelands and surprisingly see the changes on the desktop.

**To Reproduce**
There is no specific step to reproduce, just move the map with pressed right mouse button and release the button. This happens sometimes, not on every move of the map.

**Expected behavior**
If i release the button, widelands should behave normally.

**Version:**
 - OS: archlinux
 - Widelands Version: 1.1~git25843 (f83cf83@master) Debug

**Additional context**
I guess commit 8b76ebdb098847cec009b8edee27db305f985f44 is the culprit. If i checkout the commit before and compile, all works fine.

**Comment by frankystone** [comment]
Ah, forgotten: Tried with two different mice: One remote and one connected with a cable. The "locked"-button-situation happened with both mice.

**Comment by frankystone** [comment]
Try this:

1. Open the map [Forest Valley](https://www.widelands.org/maps/forest-valley/) in the editor
2. Scroll to the blue starting position (at 20/45)
3. Press right mouse button at the blue starting position and scroll to the right until the pointer is in the right of the stone barrier
4. Release the right mouse button
5. Now the bug appears for me reproducible

I guess that some other programs get opened (eg. widgets on the desktop) happens because i go wild and move the mouse fast and do right click after right click :-D

Edit: To get rid of the situation described in the bug i have to scroll back to the blue starting position and press the right mouse button once.

**Comment by frankystone** [comment]
Installed SDL-Versions here:

~~~~
$:> pacman -Qs SDL2
local/sdl2 2.0.22-2
local/sdl2_image 2.6.1-1
local/sdl2_mixer 2.6.1-1
local/sdl2_net 1:2.0.1-3
local/sdl2_ttf 2.20.0-1
~~~~

**Comment by Noordfrees** [comment]
Not reproducible for me with SDL 2.0.20 on Ubuntu Jammy. 
Are the options to grab input and use system mouse cursor enabled or disabled?

Edit: X11 or Wayland?

**Comment by frankystone** [comment]
`Grab Input` = off
`Use System Mouse Cursor` = off
`Scroll When Mouse is near Screen Edge` = off
`Invert Click and Drag Map movement direction` = off

X11

**Comment by matthiakl** [comment]
I am also able to reproduce when the right mouse button is still pressed while the cursor moves out of Widelands in window mode.
This is a regression in SDL 2.0.22 and will be fixed in the next update: https://github.com/libsdl-org/SDL/issues/5652
Try `export SDL_MOUSE_AUTO_CAPTURE=0` before starting the game and see if it fixes it.

**Comment by frankystone** [comment]
> Try `export SDL_MOUSE_AUTO_CAPTURE=0` before starting the game and see if it fixes it.

Yes that fixed it the half way. This bug does not happen if this environment variable is set. But…

A few words about my desktop setup: I have the desktop panel (the one which shows e.g. open windows) configured this way: At the bottom, Hidden, if mouse pointer comes at the bottom edge of the screen, show it.

Now this happens when widelands is running in window mode:
If i scroll the map down and release the right button, my desktop panel shows up (like the mouse pointer touches the bottom edge of the screen) although the mouse pointer is visually in the middle of the widelands window. scary…
